src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
+                cdf_start, cdf_stop = cdf_power_law(segments_blocked_nonoverlapping[i-1][2], x0, x1, α), 1.
+                push!(cdf_segments_allowed, (cdf_start, cdf_stop))
+            end
+        else
+            cdf_start, cdf_stop = cdf_power_law(segments_blocked_nonoverlapping[i-1][2], x0, x1, α), cdf_power_law(segments_blocked_nonoverlapping[i][1], x0, x1, α)
+            push!(cdf_segments_allowed, (cdf_start, cdf_stop))
+        end
+    end
+
+    return cdf_segments_allowed
+end
+
+function draw_lognormal_allowed_regions(segments_blocked::Array{Tuple{Float64,Float64},1}; μ::Float64=0., σ::Float64=1., x_min::Float64=0., x_max::Float64=Inf)
+    cdf_segments_allowed = compute_allowed_regions_cdf_lognormal(segments_blocked; μ=μ, σ=σ, x_min=x_min, x_max=x_max)
+    if length(cdf_segments_allowed) == 0
+        println("No allowed regions left to draw from; returning NaN.")
+        return NaN
+    else
+        y = draw_segmented_uniform(cdf_segments_allowed)
+        return invert_cdf_lognormal(y; μ=μ, σ=σ)
+    end
+end
+
+function draw_power_law_allowed_regions(segments_blocked::Array{Tuple{Float64,Float64},1}, x0::Float64, x1::Float64, α::Float64)
+    cdf_segments_allowed = compute_allowed_regions_cdf_power_law(segments_blocked, x0, x1, α)
+    if length(cdf_segments_allowed) == 0
+        println("No allowed regions left to draw from; returning NaN.")
+        return NaN
+    else
+        y = draw_segmented_uniform(cdf_segments_allowed)
+        return invert_cdf_power_law(y, x0, x1, α)
+    end
+end
+
+function draw_period_lognormal_allowed_regions_Hill(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; μ::Float64=0., σ::Float64=1., x_min::Float64=0., x_max::Float64=Inf, ecc::AbstractVector{Float64}=zeros(length(P)), verbose::Bool=false)
+    @assert length(P) == length(mass) == length(ecc)
+    @assert star_mass > 0
+    @assert all(P .> 0)
+    @assert all(mass .> 0)
+    @assert all(1 .> ecc .>= 0)
+
+    if length(P) > 0
+        P_segments_unstable = compute_unstable_Hill_regions_periods_given_planets(P, mass, insert_pl_mass::Float64, star_mass, sim_param; ecc=ecc, verbose=verbose)
+        P_draw = draw_lognormal_allowed_regions(P_segments_unstable; μ=μ, σ=σ, x_min=x_min, x_max=x_max)
+    else
+        P_draw = invert_cdf_lognormal(rand(Uniform(cdf_lognormal(x_min; μ=μ, σ=σ), cdf_lognormal(x_max; μ=μ, σ=σ))); μ=μ, σ=σ)
+    end
+    return P_draw
+end
+
+function draw_period_lognormal_allowed_regions_mutualHill(P::AbstractVector{Float64}, mass::AbstractVector{Float64}, insert_pl_mass::Float64, star_mass::Float64, sim_param::SimParam; μ::Float64=0., σ::Float64=1., x_min::Float64=0., x_max::Float64=Inf, ecc::AbstractVector{Float64}=zeros(length(P)), insert_pl_ecc::Float64=0., verbose::Bool=false)
+    @assert length(P) == length(mass) == length(ecc)
+    @assert 0 < insert_pl_mass < star_mass
```

Comment:
Is a zero mass planet not allowed?

Reply:
I guess there's no reason why we can't have test particles!